### DISCUSSION TITLE
NEAR CN Office Hour

### DISCUSSION THREAD

**Discussion by think-in-universe** [near-x/near-developers#4] [Ideas]
Welcome to NEAR Chinese Community [Office Hour](https://near.org/developers/get-help/office-hours/). 

Please submit your topics here if any.

**Comment by think-in-universe** [comment]
[NEAR Rust SDK 4.1.0](https://github.com/near/near-sdk-rs/releases/tag/4.1.0) `store` usage. Is it upgradable from `collection`?

**Reply by think-in-universe** [reply]
Reply from Austin:

> So first off, there is no requirement to migrate to the new collections. If you've already built a contract using them, it in most cases would not be worth the migration. I would only recommend using the `near_sdk::store` collections when starting fresh, or if the scope and size of a contract is small.
>
> There is a migration for `Vector` (which is really simple and cheap because the only state breaking change is a different length serialization) https://docs.rs/near-sdk/latest/near_sdk/collections/vector/struct.Vector.html#method.to_v2 but I'm noticing now the other ones that are possible (`UnorderedMap`, `UnorderedSet`, `LazyOption`) are not included. Let me open an issue for this now.
>
> With any collection, though, if you need, you can just use the iterator of the one you want to migrate from and any of the extend like methods on the target collection. Just be careful in the cases of `UnorderedMap`/`UnorderedSet` that there is no overlap in the prefix, as there were state-breaking changes to optimize the storage interactions for both of these.

**Comment by think-in-universe** [comment]
The `near-plugins` and `near-contract-tools` crates: 

- https://github.com/aurora-is-near/near-plugins
- https://github.com/NEARFoundation/near-contract-tools

**Comment by think-in-universe** [comment]
Browse MetaBUILD III Hackathon gallery: https://metabuild.devpost.com/project-gallery

**Comment by think-in-universe** [comment]
Developer Governance: https://near.org/developers/governance/

**Comment by think-in-universe** [comment]
Login NEAR DApp with multiple function call keys

**Comment by hsxyl** [comment]
Share about how to write contract with no std and near_sdk for minimizing the size of wasm.

**Reply by Akagi201** [reply]
near_bindgen depends on std feature right?

**Comment by think-in-universe** [comment]
Today's Topics:

1. NEAR on ETH Denver: https://neardenver.splashthat.com/
2. Developer Report: https://www.developerreport.com/
3. Interactive Learning: https://nearprimer.com/
4. NEAR Social: https://near.social/
5. Keypom: https://github.com/keypom/keypom (https://keypom.xyz/)

**Comment by think-in-universe** [comment]
### Topics: 

Blockchain + AI

- https://neardenver.splashthat.com/

How to import private key into NEAR web wallet?

1. NEAR CLI (Rust): Add a new key to account with seed phrase
2. Import account with private using Meteor Wallet
3. NEAR wallet selector: export / import accounts

**Comment by think-in-universe** [comment]
**Topics**:

bOS: Blockchain Operating System

- video by Alex: https://www.youtube.com/watch?v=3n7uhqAqsQo

near-sdk-js move out of Alpha

Rustle: uint warning not in final report

zero-balance account

**Comment by think-in-universe** [comment]
**Topics:**

- BOS: 
  - https://nearportal.com/zh-Hans/article/1325
  - https://alpha.near.org/#/

**Comment by think-in-universe** [comment]
**Topics**

- BOS
  - [technical workshop Mar 22th](https://alpha.near.org/#/calebjacob.near/widget/PostPage?accountId=root.near&blockHeight=88004806)
- Zero-Balance Account + Meta Transaction ([nearcore v1.32.0](https://github.com/near/nearcore/releases/tag/1.32.0))
  - Zero-Balance Account: NEP-448: https://github.com/near/NEPs/blob/master/neps/nep-0448.md
  - Meta Transaction: NEP-366: https://github.com/near/NEPs/blob/master/neps/nep-0366.md

**Comment by think-in-universe** [comment]
Topics

- FastAuth
    - demo: https://www.youtube.com/watch?v=dSQDKZEg3DQ&t=3494s

**Comment by think-in-universe** [comment]
Topics:

- Blockchain + AI Vision by Illia: https://alpha.near.org/calebjacob.near/widget/PostPage?accountId=root.near&blockHeight=88884689
- 2023 Q1 Protocol Roadmap Update: https://medium.com/nearprotocol/q1-protocol-roadmap-update-progress-and-changes-e2d61626e4a1 
- AstroDAO Transition: https://medium.com/nearprotocol/call-to-action-astrodao-transitions-to-the-near-community-88e8cc990d71  

BOS
- Idea: BOS Gateways Development Sprint #1 (Planning / CTA): https://alpha.near.org/devgovgigs.near/widget/gigs-board.pages.Post?id=425
- Idea: NEAR Generation ~ Dev Sprint Plan (April 3-14) #ABC https://near.social/#/devgovgigs.near/widget/gigs-board.pages.Post?id=437

**Comment by think-in-universe** [comment]
- NEAR Security Course
  - https://learnnear.club/courses/master-security-for-rust-developers-on-near-protocol/
  - https://github.com/timurguvenkaya/near-sc-security-course
  - https://twitter.com/timurguvenkaya/status/1636709981960687616?s=20

- Workgroup Update: https://gov.near.org/t/ndc-weekly-pulse-consensus-event-and-workgroup-update/34074

**Comment by think-in-universe** [comment]
BOS Example (Aurora + NEAR)
- post: https://dev.aurora.dev/posts/building-a-game-using-near-aurora-and-bos
- contract: https://github.com/aurora-is-near/aurora-contracts-sdk/blob/main/examples/tic-tac-toe/README.md
- BOS frontend: https://bos.gg/#/mob.near/widget/WidgetSource?src=nearcon.birchmd.near/widget/Aurora-Tic-Tac-Toe

Simulate JS Export from Component
- Ref: estimate swap out amount: https://near.org/near/widget/ComponentDetailsPage?src=ref-admin.near/widget/Ref.ref-swap-getEstimate
- Burrow: https://near.org/near/widget/ComponentDetailsPage?src=ciocan.near/widget/burrow-data

Complex BOS App Example
- Dev Hub: https://github.com/near/neardevhub-widgets

**Comment by think-in-universe** [comment]
BOS Feedback

- https://banyan-collective.notion.site/BOS-Feedback-Summary-Users-Builders-Partners-d3016c82a7174d0797e3d62ca886f4d6

**Comment by think-in-universe** [comment]
Meta Tx

- NEP-366: https://github.com/near/NEPs/pull/366
- Architecture: https://near.github.io/nearcore/architecture/how/meta-tx.html
- Relayer: https://github.com/near/pagoda-relayer-rs

**Comment by think-in-universe** [comment]
ZK

- Electron mainnet launch: https://medium.com/electron-labs/electron-zkbridge-1-0-mainnet-launch-3cd1e1f8f37d

NFT

- Sharddog NEARWEEK NFT: https://shard.dog/nearweek

Infrastructure

- Octopus 2.0: https://medium.com/nearprotocol/octopus-2-0-promises-to-break-barriers-between-blockchains-a64c7c80f9d1

**Comment by think-in-universe** [comment]
Messari NEAR Q2 Performance Report: https://messari.io/report/state-of-near-q2-2023

**Comment by think-in-universe** [comment]
- Web3 Hackfest Hackathon: https://web3-hackfest.devfolio.co
- Rust 合约中文教程：https://github.com/hanakannzashi/near-course-base
- BOS Web Engine v2: https://github.com/near/bos-web-engine

**Comment by think-in-universe** [comment]
- NEP-480: Account Namespaces: https://github.com/near/NEPs/pull/480
- NEP-481: Synchronous WASM Submodules: https://github.com/near/NEPs/pull/481
- NEP-485: Shared Ephemeral Storage: https://github.com/near/NEPs/pull/485

**Comment by think-in-universe** [comment]
- BOS Hack: https://www.boshacks.com/

**Comment by think-in-universe** [comment]
Update
- FastAuth SDK: https://docs.near.org/tools/fastauth-sdk
- Blockchain + AI: https://edgeintelligence.xyz/
- Dap Dap: https://twitter.com/DapDapMeUp/status/1698950078533337096

Ideas
- Developer News Letter
- Fix Awesome NEAR
- https://pikespeak.ai presentation

**Comment by think-in-universe** [comment]
Hackathon

- https://www.encode.club/encode-near-horizon-hackathon

**Comment by think-in-universe** [comment]
Hackathon
- Abstraction Hackathon: https://www.abstracting.org/

Ideas
- Chain Abstraction: https://medium.com/nearprotocol/chain-abstration-is-near-the-monolithic-and-modular-thesis-248dc51294e2

**Comment by think-in-universe** [comment]
Chain Signature

1. https://twitter.com/Cameron_Dennis_/status/1745865862450741511?s=20
    1) Slides: https://docs.google.com/presentation/d/1Bdybfz-JMwNaykzayPkPWLyfw2iQAcljufUEgLHa9sU/edit 
    2) Presentation: https://youtube.com/watch?v=rLTSBLbePLU
    3) Github Discussion: https://github.com/near/NEPs/issues/503
    4) Sample Request: https://testnet.nearblocks.io/txns/HCUw63p7f8DzGuYXGLS85nbJc62iphyeRzwXzLNPdNMP
    5) API: [https://github.com/near/mpc-recovery/blob/6f46b0b648f9cf62cddd4c4eeea7742dbab59124/contract/src/lib.rs#L291-L297](https://t.co/7bcWfOGm6z)
2. Docs: https://docs.google.com/document/d/1c2SWUo3yesoiqSDa0rzUtaKEi74MO5fYjN5U3DLdsQo/edit#heading=h.nm9ayq5wc84v
3. Testnet Contract: https://testnet.nearblocks.io/address/multichain0.testnet

**Comment by think-in-universe** [comment]
zkWASM

- Project Tracking: https://github.com/near/near-one-project-tracking/issues/6

**Comment by think-in-universe** [comment]
Chain Abstraction

- blog post: https://near.org/blog/why-chain-abstraction-is-the-next-frontier-for-web3
- twitter space: https://twitter.com/near_intern/status/1750470764376338929?s=20
- api docs: https://github.com/near/docs/issues/1677

**Comment by think-in-universe** [comment]
Chain Abstraction Demo
- SweatCoin: transfer to BNB: https://twitter.com/SweatEconomy/status/1762988475442229581
- Send ETH with NEAR Tx:
  - BOS component: https://test.near.social/md1.testnet/widget/chainsig-sign-eth-tx
  - Video: https://www.youtube.com/watch?v=0h-uY_fEoZE
  - Sepolia Faucet: https://sepoliafaucet.com/
 
NEP-519: 
- Work Group Call: https://twitter.com/NEARDevHub/status/1763674759127273603?s=20

**Comment by think-in-universe** [comment]
- Chain Signature Post by Proximity Labs: https://medium.com/@ProximityFi/a-first-look-at-chain-signatures-cross-chain-without-bridges-81c8421d153c
- NEP-519 Yield Execution: https://github.com/near/NEPs/pull/519/files
- Potlock Quadratic Funding: https://bos.potlock.org/?tab=pot&potId=build.v1.potfactory.potlock.near
- DevHub Newsletter #3: https://mailchi.mp/neardevhub/near-dev-news-3?e=d348748381

**Comment by think-in-universe** [comment]
Fast NEAR API
- https://github.com/fastnear/fastnear-api-server-rs

Chain Signature

1. [Serialization Format in JSON](https://github.com/near/near-fastauth-wallet/blob/dmd/chain_sig_docs/docs/chain_signature_api.org#serialization-format)
2. NEAR MPC Scripts
    - https://github.com/mattlockyer/mpc-script
3. NEAR Town Hall: Chain Signature Demo, transfer to accounts on BTC / ETH / BNB
    - repo: https://github.com/Pessina/near-sandbox
    - recording: https://youtu.be/XVfRuGpKMr8?t=643

BOS Web Engine
- Alpha Test: https://bwe-docs.near.dev/

**Comment by think-in-universe** [comment]
NEAR AI
- blog post: https://pages.near.org/blog/user-owned-ai-is-near/
- office hour: https://x.com/NEARProtocol/status/1796217437400269235

NEP-519: yield / resume promise
- NEP-519: https://github.com/near/NEPs/pull/519
- near-sdk-rs support: https://github.com/near/near-sdk-rs/pull/1183
- MPC draft implementation with yield / resume promise: https://github.com/near/near-sdk-rs/pull/1133/files

**Comment by think-in-universe** [comment]
OmniToken

- https://x.com/buidlnear/status/1800954913704251730
- https://hackmd.io/CpF_cEOZTkyzvDsI0SZfTw

Reclaim Protocol

- https://docs.reclaimprotocol.org/near
- https://gitlab.reclaimprotocol.org/integrations/onchain/near-sdk

BOS Web Engine

- https://github.com/near/bos-web-engine

**Comment by think-in-universe** [comment]
Chain Signature

- [[NEP-563] Deterministic Derivation Paths for MPC Network](https://github.com/near/NEPs/pull/563)

Hackathon

- [[REDACTED]](https://redacted.devpost.com/)

Ethereum Wallet on NEAR

- [Demo](https://near-examples.github.io/hello-near-examples/hello-near)

NEAR AI

- https://x.com/NEARWEEK/status/1838921964028604868

Infrastructure

- Contract Verification on NEAR Blocks: https://x.com/NEARDevHub/status/1835278293358301206 | https://nearblocks.io/address/devhub.near?tab=contract

**Comment by think-in-universe** [comment]
Multi-chain Trial Account
- https://github.com/keypom/multichain-trial-accounts

JS in Rust Contract
- Fungible Token example with customization features: https://x.com/salomonsen_p/status/1843946090967794050 | https://github.com/petersalomonsen/quickjs-rust-near/pull/14

**Comment by think-in-universe** [comment]
Ethereum Wallets on NEAR

- Demo: https://near-examples.github.io/hello-near-examples/hello-near
- NEP-518: https://github.com/near/NEPs/issues/518
- Wallet Contract: https://github.com/near/nearcore/tree/ab75966da2e4350239824605d61af5b25684e35a/runtime/near-wallet-contract 
- Wallet Selector: https://github.com/near/wallet-selector/tree/main/packages/ethereum-wallets

Omni Bridge

- Contract & Relayer: https://github.com/Near-One/omni-bridge

**Comment by think-in-universe** [comment]
Data Analytics

- NEAR Protocol live on Dune Analytics: https://dune.com/blog/near-protocol-is-now-live-on-dune
- Dashboard: https://dune.com/blockchains/near
- Data Catalog: https://docs.dune.com/data-catalog/near/overview

Use NEAR APIs in Cocos Creator

- https://github.com/xiangxn/near-api-cocos

Tutorials

- NEAR 101 Meta Tutorial: https://docs.near.org/tutorials/auction/introduction

Tooling

- near-api-rs: https://github.com/near/near-api-rs

NEAR Block Latency Test

- https://mainnet.neardata.xyz/

**Comment by think-in-universe** [comment]
Electric Capital Developer Report 2024

- Report: https://threadreaderapp.com/thread/1867295512627753432.html

**Comment by think-in-universe** [comment]
Omin Transaction Rust Library
- https://github.com/near/omni-transaction-rs

Code Injection Attack via GitHub Action
- https://blog.yossarian.net/2024/12/06/zizmor-ultralytics-injection

**Comment by think-in-universe** [comment]
NEAR AI

- NEAR AI: https://docs.near.ai/
- AI Agent Showcase: RAID: [AI Powered Token Turning TXs Into Battle Arenas](https://x.com/zacodil/status/1871297252016456104)

Infrastructure Committee

- Proposals: https://near.social/infrastructure-committee.near/widget/app?page=proposals

**Comment by think-in-universe** [comment]
NEAR Intents
- Docs: https://docs.near-intents.org/defuse-protocol
- Guide for Users and Devs: https://learnnear.club/near-intents-a-guide-for-users-and-developers/

AI Agent
- Agent PVP: https://x.com/serrrfirat/status/1875240522476032271

**Comment by think-in-universe** [comment]
AI Agent
- AI Agent Toolkits: https://www.aiagenttoolkit.xyz/#tools

Chain Signatures
- [Building Bitcoin-native dApps with NEAR’s Abstraction Stack](https://medium.com/@ProximityFi/building-bitcoin-native-dapps-with-nears-abstraction-stack-fb4aeb448eee)

**Comment by think-in-universe** [comment]
NEAR AI

- TEE on NEAR AI
  - https://x.com/near_ai/status/1883930256723304959
  - https://github.com/nearai/private-ml-sdk
- Build 1.4T parameters model: https://x.com/NEARProtocol/status/1883896255627305275
- TypeScript Agent on NEAR AI: https://x.com/zacodil/status/1885736039513551204

Chain Signatures

- Chain Signatures Roadmap: https://pages.near.org/blog/advancing-chain-signatures-whats-next/
- BTC x Chain Signatures: https://x.com/proximityfi/status/1885448472943538283

Smart Contract

- NEAR Smart Contract in Python: https://x.com/Freol/status/1884249942321230133

**Comment by think-in-universe** [comment]
NEAR Treasury: https://neartreasury.com/

RFP: FastAuth & Multichain One-Click Wallet: https://x.com/NEARProtocol/status/1888964044469436873
-  https://near.org/blog/rfps-now-open-for-fast-auth-and-multichain-one-click

AI Auditor Agent: https://near.ai/rfps/ai-auditor-agent

Increase Block Speed: https://x.com/BowenWang18/status/1890419904441893011

OmniBridge: https://x.com/NEARProtocol/status/1890470996999582043
- https://near.org/blog/omnibridge-nears-universal-solution-for-cross-chain-liquidity
- https://github.com/near-one/omni-bridge

Lantstool: NEAR RPC Tool: https://app.lantstool.dev/

**Comment by think-in-universe** [comment]
NEAR AI

- DCML: https://near.ai/blog/decentralized-confidential-machine-learning
- NEAR MCP: https://x.com/near_ai/status/1902775938720600338 | https://mcp.so/server/near-mcp/nearai

NEAR Protocol

- 8 Shards: https://x.com/NEARProtocol/status/1903780872760988037 | https://demo.neardata.xyz/

Hackathon Winners

- 1T Agents Hackathon Winners: https://x.com/i/broadcasts/1mrxmPDwVZzJy (Review: https://x.com/questflow/status/1899707488188223834 / Video: https://x.com/i/broadcasts/1mrxmPDwVZzJy)
- Useful Agents Hackathon Winners: https://x.com/near_ai/status/1902189382200914354

Developers

- Protocol Rewards Program: https://www.nearprotocolrewards.com/

**Comment by think-in-universe** [comment]
Shade Agent

- Based Names Agent: https://x.com/proximityfi/status/1906778009912447325
    - Example: https://x.com/kendaIIc/status/1907457780107346150

Hackathon

- Useful Agent Hackathon (SF): https://near.org/blog/building-useful-agents-sf-hackathon-shows-the-power-of-practical-ai
   - Details: https://x.com/near_ai/status/1902189396520206494
   - NEAR Food: [repo](https://github.com/chinesepowered/nearfood/) | [agent](https://app.near.ai/agents/chinesepowered.near/nearfood/latest)

TEE

- Intro to TEE: https://x.com/NEARWEEK/status/1907706366409506896

Projects

- Intellex: https://medium.com/intellex-ai/intellex-secures-2m-to-build-critical-infrastructure-for-decentralized-ai-on-near-protocol-337c808aad4f
- Hello Mother AI: https://x.com/hellomother_ai/status/1909229846909071363 | https://www.hellomother.ai/
- Hot Omni Token: https://omnihot.gitbook.io/omnihot-docs

FastData

- Spec: https://hackmd.io/@gJgX_4T9S4SmTnkE8aQQGQ/r1U2EOq5yx
- Demo: https://upload.fastfs.io/

NEAR AI

- Developer FAQ: https://github.com/nearai/nearai/issues/1080
- Docs Agent: https://app.near.ai/agents/gagdiez.near/docs-gpt/latest/source

**Comment by think-in-universe** [comment]
Shade Agent

- Mindshare Trading Agent: https://github.com/Yonder-Labs/mindshare_agent: ([contract](https://github.com/Yonder-Labs/mindshare_contract)) / [blogpost](https://near.org/blog/introducing-the-mindshare-index-ai-agent-autonomous-trading-powered-by-shade-agents)

NEAR AI

- NEAR MCP: https://github.com/nearai/near-mcp

**Comment by think-in-universe** [comment]
Hackathon

- ZCash x NEAR Hackathon Winners: https://x.com/NEARDevHub/status/1910379229361123537

RFP

- Chain Abstraction Gas Station: https://pages.near.org/blog/chain-abstraction-needs-gas-abstraction-rfp/

Social Graph

- NEAR Yapper Leaderboard: https://yaps.kaito.ai/near

**Comment by think-in-universe** [comment]
Chain Signatures

- EdDSA Support: [blogpost](https://pages.near.org/blog/chain-signatures-adds-eddsa-support-cross-chain-signing-for-solana-ton-stellar-sui-aptos/) | [docs](https://docs.near.org/chain-abstraction/chain-signatures/implementation)


AI

- Strawberry Computer: https://strawberry.computer/
- Tweet your idea and build UX: https://x.com/BerrryComputer/status/1920597744835862584
- websim.ai: https://websim.ai/@vgrichina/game-idea-generator
- Hackathon: Shade Agents Hackathon Submission: https://x.com/proximityfi/status/1919761483371549114

**Comment by think-in-universe** [comment]
Protocol

- [600m block time](https://pages.near.org/blog/blink-and-its-final-near-launches-600ms-blocks-and-1-2s-finality/) with [Optimistic Block](https://github.com/near/nearcore/issues/10584) ([tweet](https://x.com/BowenWang18/status/1922406962383487354))
- Global Contract: https://github.com/near/nearcore/releases/tag/2.6.1

Payment

- Coinbase x402: https://www.x402.org/ ([tweet](https://x.com/CoinbaseDev/status/1919784224170889696))
- [x402 + NEAR Intents Demo](https://github.com/think-in-universe/x402/tree/feat/nearai-intent-ts-example/examples/typescript/intents)
- [Coinbase's x402 SDK documentation](https://docs.cdp.coinbase.com/x402/docs/welcome)
- [x402 Whitepaper](https://www.x402.org/x402-whitepaper.pdf)

Hackathon

- [Shade Agent Hackathon Winners](https://x.com/proximityfi/status/1920920276591677919)

Agents

- [Mindshare Agent with 13% growth](https://x.com/proximityfi/status/1920821056488518138)

NEAR Intents

- [NEAR Intents + Smart Wallet workshop](https://github.com/shaiss/near-intent-workshop)

**Comment by think-in-universe** [comment]
Governance

- [House of Stake](https://medium.com/@nearcommunitysquad/near-governance-101-house-of-stake-378ef7aa3c0f)

Security

- [Cetus Protocol hack on Sui](https://www.coindesk.com/markets/2025/05/22/sui-networks-cetus-protocol-hit-in-apparent-hack-sending-token-prices-down-90)
  - Fix: https://github.com/CetusProtocol/integer-mate/pull/6/files
  - Audit: https://github.com/CetusProtocol/Audit/blob/main/Cetus%20Sui%20Audit%20Report%20by%20OtterSec.pdf

**Comment by think-in-universe** [comment]
Protocol

- [NEP-536: Gas refund change](https://github.com/near/NEPs/blob/master/neps/nep-0536.md)

Tooling

- [near-api-js v6.0.0](https://github.com/near/near-api-js/releases/tag/v6.0.0)

**Comment by think-in-universe** [comment]
Hackathon

- PL_GENESIS: MODULAR WORLDS HACKATHON: https://plgenesis.devspot.app/en?activeTab=overview&challenge=near
- Firehose NEAR content hackathon: https://x.com/NEARProtocol/status/1932920873956315489 (https://nearn.io/firehose/3/)

Intents

- NEAR Intents Explorer: https://explorer.near-intents.org/
- How NEAR Intents works: https://x.com/AlexAuroraDev/status/1934594296625467422/photo/1

More Markets

- Docs: https://docs.moremarkets.xyz/protocol-concepts/architecture

Tooling

- create-near-app with python: https://github.com/near/create-near-app

Agents

- Social Contest agent: https://x.com/zacodil/status/1932169318105825490

**Comment by think-in-universe** [comment]
Intents

- 101: https://x.com/NEARDevHub/status/1935747373075927123

near-api-js

- Global Contracts: https://x.com/NEARDevHub/status/1935767038712332413

Hackathon

- Agents without Master: https://x.com/proximityfi/status/1936052106323964378

**Comment by think-in-universe** [comment]
TEE

- GPU TEE: https://x.com/PhalaNetwork/status/1937875700289245346

Security

- Meta Pool security incident on mpETH contract on Ethereum: https://medium.com/meta-pool/meta-pool-security-incident-on-mpeth-contract-on-ethereum-ba7a328d45c6

Game

- near-api-cocos: https://github.com/xiangxn/near-api-cocos

**Comment by think-in-universe** [comment]
Governance

- House of Stake Frontend: https://agora-near.vercel.app/proposals

Protocol

- Mainnet Incident: https://nearone.org/postmortem-07-2025

Technologies

- FastFS: https://x.com/ekuzyakov/status/1943049950709117199

Shade Agent

- Sandbox: https://docs.near.org/ai/shade-agents/sandbox/sandbox-components

NFT Marketplace

- Hot Craft: https://hotcraft.art/

Wallet

- Intear Wallet: https://wallet.intear.tech/

**Comment by think-in-universe** [comment]
DeFi

- [Calyx](https://www.calyx.xyz/): NEAR Intents Launchpad: https://www.near.org/blog/token-season-on-near
- SUI support on NEAR Intents: https://x.com/SuiNetwork/status/1945834014910349385
- MoreMarktets launch: https://app.moremarkets.xyz/
- Unity Wallet integration with NEAR: https://x.com/unity_wallet/status/1947243192757231709
- BitGo supports NEP-141 tokens: https://x.com/BitGo/status/1945549046002905494

AI

- Phala Cloud GPU: https://x.com/PhalaNetwork/status/1945833483970224528
- NEAR AI: intro to confidential inference: https://x.com/near_ai/status/1948151984625950831
- DCML: https://x.com/NEARProtocol/status/1946574580090306791

**Comment by think-in-universe** [comment]
Infra

- Cosmos SDK runtime on NEAR: https://github.com/bpolania/NEAR-Cosmos-SDK

Intents

- Everclear: https://linktr.ee/everclearorg

AI

- Kimi K2: https://x.com/near_ai/status/1950282470907425152

**Comment by think-in-universe** [comment]
Auth

1. Privy on NEAR: https://x.com/privy_io/status/1952767998663483902 ([example](https://github.com/gagdiez/hello-privy))

Governance

1. House of Stake: https://gov.houseofstake.org/assets

AI

1. The State of AI on NEAR: https://www.near.org/blog/the-state-of-ai-on-near-2025-update
2. Bitte AI: https://x.com/near_ai/status/1953501303943884816

**Comment by think-in-universe** [comment]
AI

- IQ AI: https://app.iqai.com/
  - [NEAR MCP](https://adk.iqai.com/docs/mcp-servers/near-agent)
  - [NEAR Intents MCP](https://adk.iqai.com/docs/mcp-servers/near-intents): repo: https://github.com/IQAIcom/mcp-near-intents
  - blogpost: https://www.near.org/blog/near-and-iq-ai-partner-to-advance-tokenized-ai-agents
- Mysta: https://mysta.ai/  Browser AI Assistant
- Agents: 
  - [Bitte Snapshot Governance Agent](https://bitte.ai/chat/agentid=snapshot-agent-theta.vercel.app?agentid=snapshot-agent-theta.vercel.app): [demo](https://x.com/nategeier/status/1957744907876978765)

Newsletter

- [Attention](https://ilblackdragon.substack.com/p/introducing-attention)

Infra

- [NearBlocks API sponsorship](https://nearn.io/nearblocks/1/)

**Comment by think-in-universe** [comment]
Infrastructure

- Shelby (https://shelby.xyz/): decentralized hot storage ([post](https://www.near.org/blog/near-expands-intents-with-aptos-integration-and-shelby-decentralized-storage-partnership))

AI

- Fraction AI  (AI Agent Prediction Market): https://fractionai.xyz/dapp/alfa

**Comment by think-in-universe** [comment]
Infrastructure

- TEE-secured Price Oracle RFP: https://docs.google.com/document/d/1CUbYSw3iY_7DehgWQX2BaWZf_jbzEF1aXAdZYC4r4Ec/edit?tab=t.0

Blockchain x AI

- NEAR Protocol: Pioneering Blockchain x AI: https://x.com/NEARWEEK/status/1965689894564749649
- NEAR's positioning: https://x.com/lrettig/status/1967936759771312390

Intents

- Bug Bounty: https://x.com/HackenProof/status/1965769732562293140

AI

- Allora: https://www.allora.network/ ([github](https://github.com/allora-network/allora-chain))
- Private AI x Robotics: https://ilblackdragon.substack.com/p/keep-your-friends-close-and-your
- Fraction AI: https://fractionai.xyz/dapp

**Comment by think-in-universe** [comment]
AI

- Intents + Agents: https://ilblackdragon.substack.com/p/payments-rule-everything-around-me

DeFi

- StableFlow: https://stableflow.ai/

Governance

- House of Stake: https://gov.houseofstake.org/
  - post: https://www.near.org/blog/near-house-of-stake-is-now-live-on-mainnet

**Comment by think-in-universe** [comment]
AI

- Verifiable AI in Brave built with NEAR AI: https://brave.com/blog/browser-ai-tee/

Intents

- Intents Network Intro: https://x.com/NEARWEEK/status/1991437370118316343

DeFi

- Cross-Chain Lending: https://x.com/rhea_finance/status/1991139934296781067

Validator

- Onboarding Experience: https://x.com/Aaron_Luhning/status/1991311212160926176

Offchain Relayer

- OutLayer: https://outlayer.fastnear.com/playground
- Http Fetcher: https://x.com/NEARDevHub/status/1983889155206868996

Tooling

- NEAR Playground: https://x.com/Near_Playground/status/1970945830959055126 | https://t.co/WikvOR8wsU

RPC Infra

- RPC Latency Dashboard: https://grafana.fastnear.com/public-dashboards/577b37c6cfe84b2bae23af471d27cade
- ZAN: https://zan.top/service/apikeys

**Comment by think-in-universe** [comment]
Intents:

- Calyx Launchpad: https://calyx.xyz/
    - [source code](https://github.com/aurora-is-near/aurora-launchpad-contracts)
    - [bounty](https://hackenproof.com/programs/calyx-smart-contract)
- Dune Analytics: https://dune.com/near/near-intents

Apps

- ConsumerFi: https://www.consumerfi.ai/

OutLayer

- https://outlayer.fastnear.com/

**Comment by think-in-universe** [comment]
Protocol:

- 1M TPS: https://www.near.org/blog/near-protocol-achieves-1-million-tps
- v2.10.0 upgrade recap: https://x.com/zacodil/status/1996198993232290185

AI

- Privacy - Cloud & Chat: https://ilblackdragon.substack.com/p/your-data-can-and-will-be-used-against
- Product Launch: Cloud & Private Chat: https://x.com/NEARProtocol/status/1996220341362786538

**Comment by think-in-universe** [comment]
AI

- Integration with NEAR AI Cloud / Private Chat:
    - OLLM: https://console.ollm.com/
    - HashGraph: https://x.com/HashgraphOnline/status/1998395878303027253
  
 - Private Portable Memory
     - https://x.com/XTrace_ai/status/1998114815555026992
     
 Chain Abstraction
 
 - Hot Scan: https://hotscan.org/

**Comment by think-in-universe** [comment]
Tools

1. NEAR Stream: https://live.near.tools/
2. OutLayer Demo: https://x.com/out_layer/status/1986797317916180759
3. [Rust tooling update](https://x.com/NEARDevHub/status/1994723053067206741)

AI

1. NEAR AI Vision: https://x.com/reflexivityres/status/1988638624519172326
4. BUDIL AI Hackathon 2026: https://x.com/ekang426/status/1980320032858206683 | https://luma.com/yxrbeoyd

Education

1. NEAR Quest: https://docs.near.org/quest/introduction

Analytics

1. Ecosystem Data: https://dune.com/near/chain-abstraction-is-near

**Comment by think-in-universe** [comment]
DeFi

1. Stable Transport Protocol: https://www.stablecointransport.com/

Infrastructure

1. Infrastructure Committee 2025 Review: https://www.near.org/blog/near-infrastructure-committee-2025-year-in-review
2. OutLayer proposal: https://github.com/fastnear/near-outlayer/blob/main/ORACLE_RFP_PROPOSAL.md

**Comment by think-in-universe** [comment]
IronClaw

1. Reddit AMA: https://www.reddit.com/r/MachineLearning/comments/1rlnwsk/d_ama_secure_version_of_openclaw/
2. New releases v0.7 -> v0.15: https://x.com/ilblackdragon/status/2029640047520272507

veNEAR

1. Total Voting: https://nearblocks.io/token/venear.dao

Dev News - Feb 2026

1. https://x.com/NEARDevHub/status/2031052473318981902

**Comment by think-in-universe** [comment]
Revenue

- https://revenue.near.org/

Payment

- Machine Payments Protocol: https://x.com/NEARDevHub/status/2041458484328460735

AI

- Abound x IronClaw: https://timesofindia.indiatimes.com/business/india-business/abound-launches-ai-financial-autopilot-for-nris-in-partnership-with-near-ai/articleshow/130090515.cms

**Comment by think-in-universe** [comment]
NEARWEEK

- https://nearweek.substack.com/p/near-ecosystem-newsletter-by-nearweek-a5a

Protocol

- Preparing NEAR for the Quantum Computing Era: https://www.near.org/blog/making-near-protocol-post-quantum-safe
- FIPS-204: https://csrc.nist.gov/pubs/fips/204/final

AI 

- AttackBench: https://x.com/IronClawAI/status/2052492079188873504
- IronClaw on NEAR: Secure infrastructure for the agentic era: https://svrn.net/news/ironclaw-near-secure-ai-agent-infrastructure

DeFi

- Aurora Intents: https://intents.aurora.dev/
- Intents Widget Studio: https://x.com/auroraisnear/status/2049549379888116193
- PingPay subscriptions: https://x.com/pingpay_io/status/2049942451834507424
